Sarah afiya/feature/llm implementation

sarah-afiya opened the pull request:
The last 5 commits are of this branch (commits from Feb 15)

snadi commented on documentation_quality_analysis/analyze_library/prioritize_api/llm/detect_api_discussion.py:
```diff
@@ -11,19 +12,25 @@
+    messages=[
+        {
+            "role": "user",
+            "content": "\n\nThis is a test!",
```
are you intentionally leaving "this is a test" in the code?

snadi commented on documentation_quality_analysis/analyze_library/prioritize_api/llm/data/SO_query_examples_set2.json:
```diff
@@ -0,0 +1,12 @@
+    "api": "pandas.pivot_table",
+    "discusses_api": "yes"
+  }
+}
```
so what is the difference between set1 and set2? and why are there separate .txt files with examples? Are these meant to be used in the json files?

snadi commented on documentation_quality_analysis/analyze_library/prioritize_api/llm/messages/few_shot_learning.json:
template makes sense.

snadi commented on documentation_quality_analysis/analyze_library/prioritize_api/llm/messages/few_shot_learning_2.json:
```diff
@@ -0,0 +1,8 @@
+      "role": "user",
+      "content": ""
+    }
+  ]
```
what is this second template?

snadi commented on documentation_quality_analysis/analyze_library/prioritize_api/llm/templates/template_2.txt:
so you are doing zero shot and few shot, in both prompting styles? (json messages conversation vs. a single prompt conversation?)

snadi commented on documentation_quality_analysis/analyze_library/prioritize_api/llm/discusses_api_pipeline.py:
```diff
@@ -0,0 +1,101 @@
+
+    messages = []
+
+    if mode == Modes.ZERO_SHOT:
```
Modes --> Mode

snadi commented on documentation_quality_analysis/analyze_library/prioritize_api/llm/dtos/modes.py:
```diff
@@ -0,0 +1,7 @@
+from enum import Enum
+
+
+class Modes(Enum):
```
`class PromptMode(Enum)`

snadi reviewed (commented):
Looks good (as far as I can tell since reviewing a subset of commits is confusing)

sarah-afiya commented:
@snadi I addressed your feedback from here and also refactored the templates (now reading the templates from the template directory rather than having them hard-coded).  

I somehow messed up commits while trying to rebase, I think I fixed it now, you should see only the file changes and commits related to LLM implementation here now. Could you please give it another look and approve if it's okay?

snadi reviewed (approved):
Quickly skimmed through it and I think it matches what we discussed.

snadi commented:
and yes, I can see only the commits related to the LLM evaluation
